acceptancetests/jujupy/status.py

```diff
@@ -0,0 +1,349 @@
+
+    @property
+    def current(self):
+        return self.status.get('current')
+
+    @property
+    def version(self):
+        return self.status.get('version')
+
+    @property
+    def datetime_since(self):
+        if self.since is None:
+            return None
+        return datetime_parse(self.since)
+
+    def to_exception(self):
+        """Create an exception representing the error if one exists.
+
+        :return: StatusError (or subtype) to represent an error or None
+        to show that there is no error."""
+        if self.current not in ['error', 'failed', 'down',
+                                'provisioning error']:
+            if (self.current, self.status_name) != (
+                    'allocating', self.MACHINE):
+                return None
+        if self.APPLICATION == self.status_name:
+            return AppError(self.item_name, self.message)
+        elif self.WORKLOAD == self.status_name:
+            if self.message is None:
+                return UnitError(self.item_name, self.message)
+            elif re.match('hook failed: ".*install.*"', self.message):
+                return InstallError(self.item_name, self.message)
+            elif re.match('hook failed', self.message):
+                return HookFailedError(self.item_name, self.message)
+            else:
+                return UnitError(self.item_name, self.message)
+        elif self.MACHINE == self.status_name:
+            if self.current == 'provisioning error':
+                return ProvisioningError(self.item_name, self.message)
+            if self.current == 'allocating':
+                return StuckAllocatingError(
+                    self.item_name,
+                    'Stuck allocating.  Last message: {}'.format(self.message))
+            else:
+                return MachineError(self.item_name, self.message)
+        elif self.JUJU == self.status_name:
+            if self.since is None:
+                return AgentError(self.item_name, self.message)
+            time_since = datetime.now(tz.gettz('UTC')) - self.datetime_since
+            if time_since > AgentUnresolvedError.a_reasonable_time:
+                return AgentUnresolvedError(self.item_name, self.message,
+                                            time_since.total_seconds())
+            else:
+                return AgentError(self.item_name, self.message)
+        else:
+            raise ValueError('Unknown status:{}'.format(self.status_name),
+                             (self.item_name, self.status_value))
+
+    def __repr__(self):
+        return 'StatusItem({!r}, {!r}, {!r})'.format(
+            self.status_name, self.item_name, self.status)
+
+
+class Status:
+
+    def __init__(self, status, status_text):
+        self.status = status
+        self.status_text = status_text
+
+    @classmethod
+    def from_text(cls, text):
+        try:
+            # Parsing as JSON is much faster than parsing as YAML, so try
```

Comment:
How can we ensure the client does the "right thing" when requesting status to get json? This from_text method actually seems a bit odd at first glance. I mean that in the sense a test using our framework should use a proper handler for status right?

Reply:
Ohh, I see it's also just been moved. I'll hold my comments on what Status is actually handling :-)